pkg/datasource/util.go

```diff
@@ -109,26 +117,37 @@ func IdleTimeoutCopy(ctx context.Context, cancel context.CancelFunc, src io.Read
 		}
 	}()
 
+	var nr, nw int
+	var nws int64
+	var rErr, handleErr error
 	buf := make([]byte, DownloadBufferSize)
+	zeroByteArray := make([]byte, DownloadBufferSize)
 	for {
 		// Read will error out once the context is cancelled.
-		nr, readErr := src.Read(buf)
+		nr, rErr = src.Read(buf)
 		if nr > 0 {
-			nw, writeErr := dst.Write(buf[0:nr])
-			if writeErr != nil {
-				err = writeErr
+			// Skip writing zero data
+			if skipZeroBytes && bytes.Equal(buf[0:nr], zeroByteArray[0:nr]) {
```

Comment:
Do we really need `skipZeroBytes`? any concern to skip zero bytes always?

Reply:
@shuo-wu answered at https://github.com/longhorn/backing-image-manager/pull/38#discussion_r742492651